### ISSUE TITLE
Error Code 1: Serialization.. - When loading a serialized engine from file

### ISSUE THREAD

**Issue by giaxxi** [NVIDIA/TensorRT#1736]
I wrote a function to build and serialize an engine from a ONNX model.

When I load the file generated by the function I got the following error messages during the execution:
```
1: [stdArchiveReader.cpp::nvinfer1::rt::StdArchiveReader::StdArchiveReader::54] Error Code 1: Serialization (Serialization assertion sizeRead == static_cast<uint64_t>(mEnd - mCurrent) failed.Size specified in header does not match archive size)
4: [runtime.cpp::nvinfer1::Runtime::deserializeCudaEngine::50] Error Code 4: Internal Error (Engine deserialization failed.)
```

**Please note that** when I load an engine generated by `trtexec` or by the Python code (following the [developer guide](https://docs.nvidia.com/deeplearning/tensorrt/developer-guide/index.html#build_engine_python)) the program works fine.

**The function I'm using to parse and serialize:**

```
std::cout << "Start parsing.." << std::endl;
TRTUniquePtr<nvinfer1::IBuilder> builder{nvinfer1::createInferBuilder(trt_logger)};
uint32_t flag = 1U <<static_cast<uint32_t>(nvinfer1::NetworkDefinitionCreationFlag::kEXPLICIT_BATCH);

TRTUniquePtr<nvinfer1::INetworkDefinition> network{builder->createNetworkV2(flag)};
TRTUniquePtr<nvonnxparser::IParser> parser{nvonnxparser::createParser(*network, trt_logger)};
TRTUniquePtr<nvinfer1::IBuilderConfig> config{builder->createBuilderConfig()};

std::cout << "Parsing.." << std::endl;
if (!parser->parseFromFile(model_path.c_str(), static_cast<int>(nvinfer1::ILogger::Severity::kINFO)))
{
    std::cerr << "ERROR: could not parse the model.\n";
    return;
}

config->setMaxWorkspaceSize(1ULL << 30);
if (builder->platformHasFastFp16())
{
    config->setFlag(nvinfer1::BuilderFlag::kFP16);
}

builder->setMaxBatchSize(1);

std::cout << "Serializing.." << std::endl;
nvinfer1::IHostMemory*  serialized_engine = builder->buildSerializedNetwork(*network, *config);

std::cout << "Saving.." << std::endl;
std::ofstream p("serialized_from_cpp.engine");
p.write((const char*)serialized_engine->data(), serialized_engine->size());
p.close();
```

**System:**
 - Win 11
 -  Driver Version: 472.42
 - CUDA Version: 11.4
 - TensorRT-8.2.2.1
 - cuDNN 8.2.4

Thank you.

**Comment by aancel** [comment]
Hello,

I've encountered an issue similar to yours today, and I managed to fix it.    
On my side, I was deserializing the file without specifying to read it in binary mode.     
Loading the file while making sure to read it in binary fixed the problem:    
```
    std::ifstream file(engineFile, std::ios::in | std::ios::binary);
```

Also maybe try to revert to TensorRT version 8.0 if this doesn't work.

Hopefully this can help you sort out your issue !

Best regards
